src/lib/db/event-store.ts

```diff
@@ -217,6 +224,18 @@ class EventStore extends AnyEventEmitter implements IEventStore {
             environment: e.environment,
         };
     }
+
+    transactional(transaction: Knex.Transaction): IEventStore {
+        let clone = new (this.constructor as {
+            new (db: Knex, getLogger: LogProvider): any;
+        })(this.db, this.logProvider);
+
+        for (const attribute in this) {
+            clone[attribute] = this[attribute];
+        }
+        clone.db = transaction;
+        return clone as IEventStore;
+    }
```

Comment:
Just a random thought: we're cloning the object (this reminds me of our last week struggle with clone-deep library), and I was just thinking, what if some of the inner objects also have a reference to `db` that needs to be replaced by `transaction`? Probably, this is not a problem now, but I'm thinking if this should be recursive somehow :thinking:

Reply:
I feel we should be blocking a pull request that extends complexity in the stores like that. Part of why this is cheap is because this can be a shallow clone and I think stores that take on that kind of nested data complexity are a strong warning we should rethink our lives a little

Reply:
Yeah, we did look but it got unwieldy fast and looked relatively dangerous so we noped out of that. Ultimately we figured we'd want single block transactions at the service level and if you're slinging around existing transactions at the database level you should... probably... not do that, mostly because you're going to shoot your own foot off. But yes, very open to extending or refactoring this if you want to add more magic

Reply:
I think for now this is a good step forward, I was mostly "inspired"/biased by the way some Java frameworks handle transactions and transaction propagation: they can be defined at any level and any internal call talking to the DB (or persisting framework in this case), would be involved in the same transaction (unless explicitly you require a new transaction). But we're not solving the problem of "having a transactional framework to propagate transactions" so :+1: